src/Invitation.ts

```diff
@@ -67,6 +74,148 @@ export class InvitationV1 implements invitation.InvitationV1 {
   }
 }
 
+export class SealedInvitationHeaderV2
+  implements invitation.SealedInvitationHeaderV2
+{
+  peerHeader:
+    | invitation.SealedInvitationHeaderV2_PeerInvitationHeader
+    | undefined
+
+  selfHeader:
+    | invitation.SealedInvitationHeaderV2_SelfInvitationHeader
+    | undefined
+
+  private _sendKeyBundle: SignedPublicKeyBundleV2
+  private _inboxKeyBundle: SignedPublicKeyBundleV2
+  private _createdNs: Long
+  private _peerAddress: string | undefined
+
+  constructor({ peerHeader, selfHeader }: invitation.SealedInvitationHeaderV2) {
+    if (peerHeader && selfHeader) {
+      throw new Error(
+        'SealedInvitationHeaderV2 cannot have both peer and self header'
+      )
+    } else if (!peerHeader && !selfHeader) {
+      throw new Error('SealedInvitationHeaderV2 missing peer or self header')
+    }
+
+    this.peerHeader = peerHeader
+    this.selfHeader = selfHeader
```

Comment:
nit: would it be easier to do the conditional once e.g. `const usedHeader = peerHeader || selfHeader` and then do away with all the `||` below?